invalid-name match for a valid name in except

### Bug description

```python
C0103: Variable name "VERSION" doesn't conform to snake_case naming style (invalid-name)
This reproduces:


from importlib.metadata import PackageNotFoundError
from importlib.metadata import version

try:
    VERSION = version("ty")
except PackageNotFoundError:
    VERSION = "0.0.0"


3.3.9 does not match this.
```

### Configuration

```ini
[tool.pylint.format]
max-line-length = 120

[tool.pylint.basic]
good-names = ["i", "j", "k", "ex", "Run", "_", "f", "dn"]
extension-pkg-whitelist = ["lxml.etree", "pydantic"]

[tool.pylint."messages control"]
disable = [
    "raw-checker-failed",
    "locally-disabled",
    "suppressed-message",
    "wrong-import-order",
    "ungrouped-imports",
]

[tool.pylint.design]
max-args = 8
max-positional-arguments = 8
max-attributes = 10
min-public-methods = 0
```

### Command used

```shell
pylint a
```

### Pylint output

```python
************* Module a
a/__init__.py:9:4: C0103: Variable name "VERSION" doesn't conform to snake_case naming style (invalid-name)
```

### Expected behavior

No output.

### Pylint version

```shell
pylint 4.0.0
astroid 4.0.1
Python 3.11.12 (main, May  8 2025, 08:02:37) [GCC 13.3.0]
```

### OS / Environment

_No response_

### Additional dependencies

```python

```

--- a/doc/whatsnew/fragments/10652.false_positive
+++ b/doc/whatsnew/fragments/10652.false_positive
@@ -1,0 +1,3 @@
+Fix false positive for ``invalid-name`` on a partially uninferable module-level constant.
+
+Closes #10652

--- a/pylint/checkers/base/name_checker/checker.py
+++ b/pylint/checkers/base/name_checker/checker.py
@@ -520,9 +520,14 @@ class NameChecker(_BasicChecker):
                         self._check_name("const", node.name, node)
                 else:
                     node_type = "variable"
+                    iattrs = tuple(node.frame().igetattr(node.name))
                     if (
-                        (iattrs := tuple(node.frame().igetattr(node.name)))
-                        and util.Uninferable not in iattrs
+                        util.Uninferable in iattrs
+                        and self._name_regexps["const"].match(node.name) is not None
+                    ):
+                        return
+                    if (
+                        util.Uninferable not in iattrs
                         and len(iattrs) > 1
                         and all(
                             astroid.are_exclusive(*combo)

--- a/tests/functional/i/invalid/invalid_name/invalid_name_module_level.py
+++ b/tests/functional/i/invalid/invalid_name/invalid_name_module_level.py
@@ -1,5 +1,5 @@
 """Tests for invalid name for names declared at module level"""
-# pylint: disable=missing-class-docstring, too-few-public-methods, missing-function-docstring
+# pylint: disable=missing-class-docstring, too-few-public-methods, missing-function-docstring, wrong-import-position
 
 import collections
 
@@ -42,3 +42,12 @@ elif CONSTA:
     other_const = [2]
 else:
     other_const = [3]
+
+
+from importlib.metadata import PackageNotFoundError
+from importlib.metadata import version
+
+try:
+    VERSION = version("ty")  # uninferable
+except PackageNotFoundError:
+    VERSION = "0.0.0"

